controllers/usersignup/usersignup_controller.go

```diff
@@ -131,7 +131,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	if originalUserSignupName, found := userSignup.Annotations[migrationReplacesAnnotationName]; found {
 		return r.cleanupMigration(userSignup, originalUserSignupName, request, logger)
 	}
-
+	if labels := userSignup.GetLabels(); labels == nil {
```

Comment:
you can make it simpler :-)  
```suggestion
	if userSignup.GetLabels() == nil {
```